website/content/docs/secrets/openldap.mdx

```diff
@@ -172,11 +176,125 @@ only Vault will know.
+password and enable the account.
+
+- Passwords in AD are set using the `unicodePwd` field. This must be proceeded by two (2) colons (`::`).
+- When setting a password programatically in AD, the following critera must be met:
+  - The password must be enclosed in double quotes (`" "`)
+  - The password must be in [`UTF16LE` format](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/6e803168-f140-4d23-b2d3-c3a8ab5917d2)
+  - The password must be `base64`-encoded
+  - Additional details can be found [here](https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/set-user-password-with-ldifde)
+
+- Once a user's password has been set, it can be enabled. AD uses the `userAccountControl` field for this purpose:
+  - To enable the account, set `userAccountControl` to `512`
+  - You will likely also want to disable AD's password expiration for this dynamic user account. The
+`userAccountControl` value for this is: `65536`
+  - `userAccountControl` flags are cumulative, so to set both of the above two flags, add up the two values
+(`512 + 65536 = 66048`): set `userAccountControl` to `66048`
+  - See [here](https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/useraccountcontrol-manipulate-account-properties#property-flag-descriptions)
+for details on `userAccountControl` flags
+
+`sAMAccountName` is a common field when working with AD users. It is used to provide compatibility with legacy
+Windows NT systems and has a limit of 20 characters. Keep this in mind when defining your `username_template`.
+See [here](fhttps://docs.microsoft.com/en-us/windows/win32/adschema/a-samaccountname) for additional details.
+
+With regard to adding dynamic users to groups, AD doesn't let you directly modify a user's `memberOf` attribute.
+The `member` attribute of a group and `memberOf` attribute of a user are
+[linked attributes](https://docs.microsoft.com/en-us/windows/win32/ad/linked-attributes). Linked attributes are
+forward link/back link pairs, with the forward link able to be modified. In the case of AD group membership, the
+group's `member` attribute is the forward link. In order to add a newly-created dynamic user to a group, we also
+need to issue a `modify` request to the desired group and update the group membership with the new user.
+
+#### Active Directory LDIF Example
+
+The various `*_ldif` parameters are templates that use the [go template](https://golang.org/pkg/text/template/)
+language. A complete LDIF example for creating an Active Directory user account is provided here for reference:
+
+```ldif
+dn: CN={{.Username}},OU=HashiVault,DC=adtesting,DC=lab
+changetype: add
+objectClass: top
+objectClass: person
+objectClass: organizationalPerson
+objectClass: user
+userPrincipalName: {{.Username}}@adtesting.lab
+sAMAccountName: {{.Username}}
+
+dn: CN={{.Username}},OU=HashiVault,DC=adtesting,DC=lab
+changetype: modify
+replace: unicodePwd
+unicodePwd::{{ printf "%q" .Password | utf16le | base64 }}
```

Comment:
I didn't see `printf` in the list of functions above. If it is the case that any of the Go template functions are also available, it is probably good to highlight that fact at the start of the template function list and link to the list of functions in Go docs.

Reply:
It is available through the template language (not a custom one from us). I'll add a note about this.